### DISCUSSION TITLE
How can i set the latest GIT Hash as env variable?

### DISCUSSION THREAD

**Discussion by Lucas-Gruber** [coollabsio/coolify#1553] [Q&A] [answered]
The Question is in the title.

Vercel example: 
![CleanShot 2023-12-16 at 2  14 16@2x](https://github.com/coollabsio/coolify/assets/72545889/bc8609ca-26e0-472b-9869-63110baec98c)

**Comment by andrasbacsai** [comment]
Hey, it is set in `SOURCE_COMMIT` env variable in most of the build packs. Which one you would like to use?

**Reply by Lucas-Gruber** [reply]
I currently using Nixpacks with an NuxtJS 3 app.
When i try to access the var: `process.env.SOURCE_COMMIT` unfortunately it doesnt work. And when i execute `printenv` in the ui i become this output:
```shell
NUXT_PUBLIC_FILE_BASE=xxx
LIBRARY_PATH=~/.nix-profile/lib:
NPM_CONFIG_PRODUCTION=
USER=root
CI=
HOSTNAME=673f9019ce8b
GIT_SSL_CAINFO=/etc/ssl/certs/ca-certificates.crt
LD_LIBRARY_PATH=/usr/lib
NUXT_PUBLIC_URL=xxx
NUXT_CMS_TOKEN=xxx
PORT=3000
HOME=/root
QTDIR=~/.nix-profile:
NIXPKGS_ALLOW_BROKEN=1
ENV=/etc/profile
NIX_PATH=/nix/var/nix/profiles/per-user/root/channels
NIXPKGS_ALLOW_INSECURE=1
NIXPKGS_ALLOW_UNFREE=1
PATH=/nix/var/nix/profiles/default/bin:/nix/var/nix/profiles/default/sbin:/bin:/sbin:/usr/bin:/usr/sbin
NIX_SSL_CERT_FILE=/etc/ssl/certs/ca-certificates.crt
NIXPACKS_PATH=/app/node_modules/.bin:
NIXPACKS_METADATA=
PWD=/app
CPATH=~/.nix-profile/include:
NUXT_AUTH_SECRET=xxx
NODE_ENV=STAGING
```

**Reply by andrasbacsai** [reply, marked as answer]
Oh, I know why. Fixing!

**Reply by Lucas-Gruber** [reply]
Thank you very much! Is then the SOURCE_COMMIT available?

**Reply by andrasbacsai** [reply]
> Thank you very much! Is then the SOURCE_COMMIT available?

In the next release, yes.

**Reply by Lucas-Gruber** [reply]
> > Thank you very much! Is then the SOURCE_COMMIT available?
> 
> In the next release, yes.

Okay, thank you very much!

**Comment by andrasbacsai** [comment]
> > > Thank you very much! Is then the SOURCE_COMMIT available?
> > 
> > 
> > In the next release, yes.
> 
> Okay, thank you very much!

I was off for a few days (🎄), but released a new version that has this.

**Reply by Lucas-Gruber** [reply]
Thank you and no problem Christmas is important ^^

Unfortunately i cant try this because i cannot deploy my app since the new update. ( #1588 )

**Reply by Lucas-Gruber** [reply]
Okay i reinstalled my VM and now it works. I have now the Problem that i have to make something like: 
`NUXT_PUBLIC_COMMIT_HASH=$SOURCE_COMMIT` in env vars because nuxt 3 cannot access env vars without the NUXT_PUBLIC-Prefix. When enter this in the env vars it doesnt work. It says the var is null.

Then i tried it with following start command: 
`NUXT_PUBLIC_COMMIT_HASH=$SOURCE_COMMIT && node .output/server/index.mjs` or
`NUXT_PUBLIC_COMMIT_HASH=$SOURCE_COMMIT; node .output/server/index.mjs` or
`NUXT_PUBLIC_COMMIT_HASH=$SOURCE_COMMIT node .output/server/index.mjs`

But also, the env var is null. And in the Deployment logs comes following: 
```bash
╔════════════════════════ Nixpacks v1.20.0 ════════════════════════╗
║ setup │ nodejs_18, yarn-1_x ║
║──────────────────────────────────────────────────────────────────║
║ install │ yarn install --frozen-lockfile ║
║──────────────────────────────────────────────────────────────────║
║ build │ yarn run build ║
║──────────────────────────────────────────────────────────────────║
║ start │ NUXT_PUBLIC_TEST4= && node .output/server/index.mjs ║
╚══════════════════════════════════════════════════════════════════╝
```

have you any idea how i can fix this?

**Reply by andrasbacsai** [reply]
In the next version, I will add support for this. All you need to do is define your variables like this:
![image](https://github.com/coollabsio/coolify/assets/5845193/38a5b26c-91d8-4857-9c3c-8682f8be49a5)

**Reply by andrasbacsai** [reply]
Also added this section: https://coolify.io/docs/environment-variables#predefined-variables
